lib/vssapi-sys/src/lib.rs

```diff
@@ -2,19 +2,53 @@
 // Licensed under the MIT License <LICENSE.md>
 //! FFI bindings to vssapi.
 #![cfg(windows)]
+#![allow(non_snake_case)]
 extern crate winapi;
 use winapi::*;
 
-// x86_64 will work once rust issue 23216 is fixed.
-// Starting link_name with \x01 prevents the name from being mangled further.
 #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
 extern "system" {
-    #[cfg(target_arch = "x86_64")]
-    #[link_name="\x01?CreateVssBackupComponents@@YAJPEAPEAVIVssBackupComponents@@@Z"]
-    pub fn CreateVssBackupComponents(ppBackup: *mut *mut IVssBackupComponents) -> HRESULT;
-    
-    #[cfg(target_arch = "x86")]
-    #[link_name="\x01?CreateVssBackupComponents@@YGJPAPAVIVssBackupComponents@@@Z"]
-    pub fn CreateVssBackupComponents(ppBackup: *mut *mut IVssBackupComponents) -> HRESULT;
+    pub fn CreateVssBackupComponentsInternal(ppBackup: *mut *mut IVssBackupComponents) -> HRESULT;
+    pub fn CreateVssExamineWriterMetadataInternal(
+        bstrXML: BSTR, ppMetadata: *mut *mut IVssExamineWriterMetadata
+    ) -> HRESULT;
+    pub fn IsVolumeSnapshottedInternal(
+        pwszVolumeName: VSS_PWSZ, pbSnapshotsPresent: *mut BOOL, plSnapshotCapability: *mut LONG
+    ) -> HRESULT;
+    pub fn VssFreeSnapshotPropertiesInternal(pProp: *mut VSS_SNAPSHOT_PROP);
+    pub fn GetProviderMgmtInterfaceInternal(
+        ProviderId: VSS_ID, InterfaceId: IID, ppItf: *mut *mut IUnknown
+    ) -> HRESULT;
+    pub fn ShouldBlockRevertInternal(wszVolumeName: LPCWSTR, pbBlock: *mut bool) -> HRESULT;
 }
 
+#[inline]
+pub unsafe fn CreateVssBackupComponents(ppBackup: *mut *mut IVssBackupComponents) -> HRESULT {
+    CreateVssBackupComponentsInternal(ppBackup)
+}
+#[inline]
+pub unsafe fn CreateVssExamineWriterMetadata(
+    bstrXML: BSTR, ppMetadata: *mut *mut IVssExamineWriterMetadata
+) -> HRESULT {
+    CreateVssExamineWriterMetadataInternal(bstrXML, ppMetadata)
+}
+#[inline]
+pub unsafe fn IsVolumeSnapshotted(
+    pwszVolumeName: VSS_PWSZ, pbSnapshotsPresent: *mut BOOL, plSnapshotCapability: *mut LONG
+) -> HRESULT {
+    IsVolumeSnapshottedInternal(pwszVolumeName, pbSnapshotsPresent, plSnapshotCapability)
+}
+#[inline]
+pub unsafe fn VssFreeSnapshotProperties(pProp: *mut VSS_SNAPSHOT_PROP) {
+    VssFreeSnapshotPropertiesInternal(pProp)
+}
+#[inline]
+pub unsafe fn GetProviderMgmtInterface(
+    ProviderId: VSS_ID, InterfaceId: IID, ppItf: *mut *mut IUnknown
+) -> HRESULT {
+    GetProviderMgmtInterfaceInternal(ProviderId, InterfaceId, ppItf)
+}
+#[inline]
+pub unsafe fn ShouldBlockRevert(wszVolumeName: LPCWSTR, pbBlock: *mut bool) -> HRESULT {
+    ShouldBlockRevertInternal(wszVolumeName, pbBlock)
+}
```

Comment:
Could you leave out these inline functions for now? I want the `-sys` crates to just be extern declarations for now.